code/modules/food/kitchen/microwave.dm

```diff
@@ -409,6 +466,32 @@
 		if ("dispose")
 			dispose()
 	return
+	
+/obj/machinery/microwave/verb/Eject()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Eject content"
+	usr.visible_message(
+	"<span class='notice'>[usr] tries to open [src] and remove its contents.</span>" ,
+	"<span class='notice'>You try to open [src] and remove its contents.</span>"
+	)
+
+	if(!do_after(usr, 1 SECONDS, target = src))
+		return
+
+	usr.visible_message(
+	"<span class='notice'>[usr] opened [src] and has taken out [english_list(contents)].</span>" ,
+	"<span class='notice'>You have opened [src] and taken out [english_list(contents)].</span>"
+	)
+	dispose()
+
+/obj/machinery/microwave/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(!mover)
+		return 1
+	if(mover.checkpass(PASSTABLE))
+	//Animals can run under them, lots of empty space
+		return 1
+	return ..()
 
 /obj/machinery/microwave/advanced // specifically for complex recipes
 	name = "deluxe microwave"

```

Comment:
By the way, does deluxe microwave still have a reason to exist?
I also don't see any edits to thecake.dm, so is chaos cake recipe tested to still be functional/doable in new cooking system? Would it even require a deluxe microwave with new system?

Reply:
Honestly? Deluxe Microwave doesn't really have a reason to exist, but redoing the entire Chaos Cake to be made in the oven is something I'd rather do later. AFAIK it's still doable with the current microwave.

Ideally? You'd make each layer in the oven, etc.

Reply:
Marked this as a TODO and will be done later (in a separate PR)